### PR TITLE
Avoid item names

### PR DESCRIPTION
**Pull request by brendanzab**
Noticed that the item names weren’t being checked during fresh name generation.

### CODE REVIEW

**Comment by Kmeakin** [comment]
Could you add some tests to demonstrate cases where fresh local names were clashing with item names?

**Comment by brendanzab** [comment]
> Could you add some tests to demonstrate cases where fresh local names were clashing with item names?

Good idea! Was a bit lazy.

File: tests/succeed/distillation/fresh-names.snap [hunk 1]
```diff
@@ -0,0 +1,12 @@
+stdout = '''
+def a : () : Type = ();
+def const1 : fun (@A : Type) (@B : Type) -> A -> B -> A =
+fun @b @c x y => let x1 : b = x;
```
Comment (brendanzab) [inline tests/succeed/distillation/fresh-names.snap:4]: I wonder if we could actually pull these binder names from the type somewhere?

Comment (Kmeakin) [reply tests/succeed/distillation/fresh-names.snap:4]: What do you mean?

File: tests/succeed/distillation/fresh-names.snap [hunk 2]
```diff
@@ -0,0 +1,12 @@
+stdout = '''
+def a : () : Type = ();
```
Comment (Kmeakin) [inline tests/succeed/distillation/fresh-names.snap:2]: Looks like the type annotation on this item is being distilled in check mode instead of synth mode?

Comment (brendanzab) [reply tests/succeed/distillation/fresh-names.snap:2]: Oooh, good catch

File: tests/succeed/distillation/fresh-names.snap (hunk 1 above)
Comment (brendanzab) [reply tests/succeed/distillation/fresh-names.snap:4]: Like, try to use `A` and `B`. Just a thought though!

**Pull request merged**